### ISSUE TITLE
Will tf.python.control_flow_ops.cond in BatchNormLayer be deprecated?

### ISSUE THREAD

**Issue by boscotsang** [zsdonghao/tensorlayer#56]
In latest tensorflow api, tf.cond is available while there is no tf.python.control_flow_ops.cond. I'm not sure that if the tf.cond is available and tf.python.control_flow_ops.cond behave well in tensorflow 0.12. I'm wondering if there is a need to change the tf.python.control_flow_ops.cond to tf.cond. What's your opinion?

**Comment by zsdonghao** [comment]
@boscotsang Thank you for your advice, please do that.

BTW, I am updating BatchNormLayer for TensorFlow 0.12, please check the lastest version in Github.

https://github.com/zsdonghao/tensorlayer/issues/57

**Comment by boscotsang** [comment]
@zsdonghao How can I share the variables of BN in training and evaluating phase to evaluate while training? Besides, it seems that the mean and variance of test phase should take the mean and variance estimated during training because during the test phase there may be only one image so that it's hard to estimate the mean and variance of one image.

**Comment by zsdonghao** [comment]
@boscotsang 
I have two implementation at the moment : `BatchNormLayer` and `BatchNormLayer5` in https://github.com/zsdonghao/tensorlayer/blob/master/tensorlayer/layers.py

Now, I save the beta and gamma only. Yes, I agree with you, " it seems that the mean and variance of test phase should take the mean and variance estimated during training " instead of "using mean and variance of the testing data".

**Comment by zsdonghao** [comment]
@boscotsang the lastest implementation use the fixed mean and variance of training data.

**Comment by boscotsang** [comment]
@zsdonghao It works well.Thank you.

**Comment by zsdonghao** [comment]
Automatically closing due to lack of recent activity. Please reopen when additional information becomes available. Thanks!
